pkg/uploader/provider/restic.go

```diff
@@ -13,22 +13,280 @@ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 See the License for the specific language governing permissions and
 limitations under the License.
 */
+
 package provider
 
 import (
+	"bytes"
+	"context"
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 	v1 "k8s.io/api/core/v1"
 
 	"github.com/vmware-tanzu/velero/internal/credentials"
 	velerov1api "github.com/vmware-tanzu/velero/pkg/apis/velero/v1"
+	"github.com/vmware-tanzu/velero/pkg/restic"
+	"github.com/vmware-tanzu/velero/pkg/uploader"
+	"github.com/vmware-tanzu/velero/pkg/util/exec"
+	"github.com/vmware-tanzu/velero/pkg/util/filesystem"
 )
 
+// BackupFunc mainly used to make testing more convenient
+var ResticBackupFunc = restic.BackupCommand
+var ResticRunCMDFunc = exec.RunCommand
+var ResticGetSnapshotSizeFunc = restic.GetSnapshotSize
+var ResticGetVolumeSizeFunc = restic.GetVolumeSize
+var ResticRestoreFunc = restic.RestoreCommand
+
+type resticProvider struct {
+	repoIdentifier  string
+	credentialsFile string
+	caCertFile      string
+	repoEnv         []string
```

Comment:
Better to remove it as `cmdEnv`, this env is not seen anything specific connection to a repository

Reply:
Do you mean to rename it to cmdEnv? If so, I've renamed it